lib/src/render/divider/divider_shortcut_event.dart

```diff
@@ -0,0 +1,72 @@
+import 'package:appflowy_editor/appflowy_editor.dart';
+import 'package:flutter/material.dart';
+import 'divider_node_widget.dart';
+
+// insert divider into a document by typing three minuses.
+// ---
+ShortcutEvent insertDividerEvent = ShortcutEvent(
+  key: 'Divider',
+  character: '-',
+  handler: _insertDividerHandler,
+);
+
+ShortcutEventHandler _insertDividerHandler = (editorState, event) {
+  final selection = editorState.service.selectionService.currentSelection.value;
+  final textNodes = editorState.service.selectionService.currentSelectedNodes
+      .whereType<TextNode>();
+  if (textNodes.length != 1 || selection == null) {
+    return KeyEventResult.ignored;
+  }
+  final textNode = textNodes.first;
+  if (textNode.toPlainText() != '--') {
+    return KeyEventResult.ignored;
+  }
+  final transaction = editorState.transaction
+    ..deleteText(textNode, 0, 2) // remove the existing minuses.
+    ..insertNode(textNode.path, Node(type: kDividerType)) // insert the divder
+    ..afterSelection = Selection.single(
+      // update selection to the next text node.
+      path: textNode.path.next,
+      startOffset: 0,
+    );
+  editorState.apply(transaction);
+  return KeyEventResult.handled;
+};
```

Comment:
I am thinking to allow this to be used for multiple different characters, we can change it like this:
```suggestion
ShortcutEventHandler _insertDividerHandler = (editorState, event) {
  final character = event?.character;
  if (character == null) {
    return KeyEventResult.ignored;
  }
  
  final selection = editorState.service.selectionService.currentSelection.value;
  final textNodes = editorState.service.selectionService.currentSelectedNodes
      .whereType<TextNode>();
  if (textNodes.length != 1 || selection == null) {
    return KeyEventResult.ignored;
  }
  
  final textNode = textNodes.first;
  if (textNode.toPlainText() != (character * 2)) {
    return KeyEventResult.ignored;
  }
  
  final transaction = editorState.transaction
    ..deleteText(textNode, 0, 2) // remove the existing characters.
    ..insertNode(textNode.path, Node(type: kDividerType)) // insert the divider
    ..afterSelection = Selection.single( // update selection
      path: textNode.path.next,
      startOffset: 0,
    );
  editorState.apply(transaction);

  return KeyEventResult.handled;
};
```

Reply:
I suggest we can add ```if``` block to check if character is ```'-'```,```'*'```,```'_'``` and only then insert the divider .

Reply:
Why? The event handler will be hooked to a certain character, and so that will decide which is allowed.

Reply:
Ok no need then.